src/main/java/athena/Storage.java

```diff
@@ -0,0 +1,91 @@
+package athena;
+import java.io.*;
+
+
+public class Storage {
+    private String filePath;
+    private taskList tasks;
+
+    /**
+     * methods needed from tasklist
+     * Name
+     * Time
+     * Duration
+     * Deadline
+     * Recurrence
+     * Importance
+     * Notes
+     * Index
+     * Restore
+     * task and tasklist constructor
+     * <p>
+     * methods given to tasklist
+     * saveTaskListData(index) if index -1 then save everything
+     * loadTaskListData return taskList and highest index(-1 if no existing data)
+     * <p>
+     * methods needed from Timetable
+     * date
+     * time
+     * task
+     * index
+     * <p>
+     * methods given to Timetable
+     * saveTimetableData()
+     * loadTimetableData return timetable
+     * <p>
+     * methods given to UI
+     * lastSaved
+     * index
+     * <p>
+     * methods given to Parser
+     * ???
+     */
+
+    private int size;
+
+    public Storage(String filepath) {
+        this.filePath = filepath;
+
+    }
+    public void saveTaskListData(taskList tasks) {
+        this.tasks=tasks;
+        try {
+            FileWriter csvWriter = new FileWriter(filePath);
+            for(String task : tasks.getList()) {
+                csvWriter.append(task+","+"\n");
+            }
+            csvWriter.close();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+
+    }
+
+
+    public taskList loadTaskListData() {
+        File csvFile = new File(filePath);
+        taskList output = null;
+        if (csvFile.isFile()) {
```

Comment:
I think it would be better to have

```
if (!csvFile.isFile()) {
    // handle error
    return null;
}
```

so that we don't have all the main logic inside an if-block.